### PR TITLE
fixed adding/removing routes when split tunneling is disabled

### PR DESCRIPTION
**Pull request by Nethius**

### CODE REVIEW

File: client/ui/controllers/settingsController.cpp [hunk 1]
```diff
@@ -144,7 +146,12 @@ void SettingsController::clearSettings()
     m_serversModel->resetModel();
     m_languageModel->changeLanguage(
             static_cast<LanguageSettings::AvailableLanguageEnum>(m_languageModel->getCurrentLanguageIndex()));
-    m_sitesModel->setRouteMode(Settings::RouteMode::VpnAllSites);
+
+    m_sitesModel->setRouteMode(Settings::RouteMode::VpnOnlyForwardSites);
```
Comment (outspace) [inline client/ui/controllers/settingsController.cpp:150]: Why we use VpnOnlyForwardSites by default with Split Tunneling false? 
VpnAllSites looks better in this case.

Comment (Nethius) [reply client/ui/controllers/settingsController.cpp:150]: because Settings::RouteMode::VpnAllSites is not needed in the ui

**Review by pokamest — APPROVED** [review]

File: client/vpnconnection.cpp [hunk 2]
```diff
@@ -65,32 +65,38 @@ void VpnConnection::onConnectionStateChanged(Vpn::ConnectionState state)
...
 
             if (!m_vpnConfiguration.value(config_key::configVersion).toInt()) {
-                if (m_settings->routeMode() != Settings::VpnAllSites) {
-                    IpcClient::Interface()->routeDeleteList(m_vpnProtocol->vpnGateway(), QStringList() << "0.0.0.0");
-                    // qDebug() << "VpnConnection::onConnectionStateChanged :: adding custom routes, count:" << forwardIps.size();
-                }
                 QString dns1 = m_vpnConfiguration.value(config_key::dns1).toString();
                 QString dns2 = m_vpnConfiguration.value(config_key::dns2).toString();
 
+                qDebug() << NetworkUtilities::ipAddressFromIpWithSubnet(dns1);
```
Comment (outspace) [inline client/vpnconnection.cpp:71]: please cleanup this

File: client/vpnconnection.cpp [hunk 3]
```diff
@@ -302,36 +308,34 @@ void VpnConnection::appendSplitTunnelingConfig()
...
+        auto sites = m_settings->getVpnIps(routeMode);
+        for (const auto &site : sites) {
+            sitesJsonArray.append(site);
+        }
 
-    // Allow traffic to Amezia DNS
-    if (routeMode == Settings::VpnOnlyForwardSites) {
-        sitesJsonArray.append(m_vpnConfiguration.value(config_key::dns1).toString());
-        sitesJsonArray.append(m_vpnConfiguration.value(config_key::dns2).toString());
+        // Allow traffic to Amezia DNS
```
Comment (outspace) [inline client/vpnconnection.cpp:320]: Amnezia DNS

File: client/ui/controllers/settingsController.cpp (hunk 1 above)
Comment (outspace) [reply client/ui/controllers/settingsController.cpp:150]: please check https://github.com/amnezia-vpn/amnezia-client/pull/744/files#diff-2a5303f3e078360a1dbb1ff0eda7b8d3da2268a2e39aee2f08de98c99206379dR76

**Pull request merged**